cloudinit/config/cc_apt_configure.py

```diff
@@ -515,7 +519,7 @@ def get_apt_cfg() -> Dict[str, str]:
     }
 
 
-def generate_sources_list(cfg, release, mirrors, cloud):
+def generate_sources_list(cfg, release, mirrors, cloud, keys={}):
```

Comment:
We should remove the default here to avoid default mutable parameters. Since there's only one call point in this file, I don't think there's any issue making this 
```python
def generate_sources_list(cfg, release, mirrors, cloud, keys):
```
I only see [one test call point](https://github.com/canonical/cloud-init/blob/main/tests/unittests/config/test_cc_apt_configure.py#L331), so if we update that to pass `{}`, we should be good.